### PR TITLE
feat: CollectionTtl to pass to collection methods.

### PR DESCRIPTION
**Pull request by schwern**
Here is the promised CollectionTtl. I'm hoping to get it merged so we can all start using it.

The docs are copied from dotnet and have not yet been converted.

### CODE REVIEW

**Review by virratanasangpunth — APPROVED** [review]
LGTM

Looks like you might need to change the commit message and/or PR title: `Error: Unknown release type "feature" found in pull request title "feature: CollectionTtl to pass to collection methods.". `

File: src/utils/collection-ttl.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  public static of(ttlSeconds: number): CollectionTtl {
+    return new CollectionTtl(ttlSeconds, true);
+  }
+
+  /** Constructs a <see cref="CollectionTtl"/> with the specified <see cref="TimeSpan"/>.
+   *  Will only refresh if the TTL is provided (ie not <see langword="null" />).
+   */
+  public static refreshTtlIfProvided(
+    ttlSeconds: number | null = null
+  ): CollectionTtl {
```
Comment (virratanasangpunth) [inline src/utils/collection-ttl.ts:58]: you gonna update the docs in a subsequent PR?

**Review by virratanasangpunth — DISMISSED** [review]

File: src/utils/collection-ttl.ts (hunk 1 above)
Comment (schwern) [reply src/utils/collection-ttl.ts:58]: I'm going to do it now.

File: src/utils/collection-ttl.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    return new CollectionTtl(ttlSeconds, true);
+  }
+
+  /** Constructs a CollectionTtl with the specified TTL.
+   *  Will only refresh if the TTL is provided.
+   * @constructor
+   * @returns {CollectionTtl}
+   */
+  public static refreshTtlIfProvided(
+    ttlSeconds: number | null = null
```
Comment (virratanasangpunth) [inline src/utils/collection-ttl.ts:80]: You don't need to `@param` this in the doc?

**Review by virratanasangpunth — APPROVED** [review]

**Pull request merged**